server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueViewModule.java

```diff
@@ -86,8 +88,9 @@
             .addPartitionKey(EnqueuedMailsTable.QUEUE_NAME, text())
             .addPartitionKey(EnqueuedMailsTable.TIME_RANGE_START, timestamp())
             .addPartitionKey(EnqueuedMailsTable.BUCKET_ID, cint())
-            .addClusteringColumn(EnqueuedMailsTable.MAIL_KEY, text())
+            .addClusteringColumn(EnqueuedMailsTable.ENQUEUE_ID, timeuuid())
```

Comment:
are you sure timeuuid is the same thing as UUID.timeBased? Also, the coupling looks brittle: what if the UUID is created by another function? You can't enforce it's nature by its type. I would rather use a simple uuid here.

Reply:
Well I remembered within a cassandra scope we tended to always favor time based uuid as the time ordering might enhance debugging. However here we are in a slicing context.

To be honnest itmakes no difference with me.

Reply:
The uuid type stored is not the same thing as the declared schema. It's cool if we have time based uuid but we probably don't what to break things when it's not the case.